### PR TITLE
Check if the ref shader is valid in visual shader's `_update_option_menu`

### LINKED ISSUE #87337: Dragging a connection from the input port in a visual shader results in an error
### Tested versions

Reproduceable on `master` at 1952f64b07b2a0d63d5ba66902fd88190b0dcf08.

### System information

Godot v4.3.dev (1952f64b0) - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3070 Ti Laptop GPU (NVIDIA; 31.0.15.3742) - 12th Gen Intel(R) Core(TM) i7-12700H (20 Threads)

### Issue description

Dragging a new connection from an input port and dropping it to create a new node results in an error in the visual shader plugin.

```
ERROR: Condition "!shader_rid.is_valid()" is true. Returning: 0
   at: VisualShaderNodeParameterRef::get_parameters_count (scene\resources\visual_shader.cpp:3605)
scene\resources\visual_shader.cpp:3605 - Condition "!shader_rid.is_valid()" is true. Returning: 0
```

I couldn't reproduce this by dragging from an output port. Also, on a few occasions there was no error even for the input port, but otherwise it was reliably reproduceable.

### Steps to reproduce

https://github.com/godotengine/godot/assets/11782833/7de25012-fa40-4fc6-a78a-7a04a262ea2d

### Minimal reproduction project (MRP)

N/a

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes https://github.com/godotengine/godot/issues/87337

### CODE REVIEW

File: scene/resources/visual_shader.h [hunk 1]
```diff
@@ -639,6 +639,7 @@ class VisualShaderNodeParameterRef : public VisualShaderNode {
 	virtual PortType get_output_port_type(int p_port) const override;
 	virtual String get_output_port_name(int p_port) const override;
 
+	bool get_shader_valid() const;
```
Comment (AThousandShips) [inline scene/resources/visual_shader.h:642]:
```suggestion
	bool is_shader_valid() const;
```
I'd use this naming, `get_shader_valid` sounds like `return shader.is_valid() ? shader : RID()` to me

Comment (jsjtxietian) [reply scene/resources/visual_shader.h:642]: Thanks!

**Review by paddy-exe — APPROVED** [review]
Works as expected and fixes the bug

https://github.com/godotengine/godot/assets/38077837/8754a5a4-42ca-477d-9319-cb910d02b1ff

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**